core/src/main/java/com/exadel/aem/toolkit/core/injectors/RequestParamInjector.java

```diff
@@ -0,0 +1,78 @@
+package com.exadel.aem.toolkit.core.injectors;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.api.request.RequestParameterMap;
+import org.apache.sling.models.spi.DisposalCallbackRegistry;
+import org.apache.sling.models.spi.Injector;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.exadel.aem.toolkit.core.injectors.annotations.RequestParam;
+
+@Component(service = Injector.class,
+    property = Constants.SERVICE_RANKING + ":Integer=" + Integer.MAX_VALUE
+)
+public class RequestParamInjector implements Injector {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RequestParamInjector.class);
+    public static final String NAME = "eak-request-parameter-injector";
+
+    @Nonnull
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+
+    @CheckForNull
+    @Override
+    public Object getValue(final @Nonnull Object adaptable,
+                           final String name,
+                           final @Nonnull Type type,
+                           final @Nonnull AnnotatedElement element,
+                           final @Nonnull DisposalCallbackRegistry callbackRegistry) {
+
+        final RequestParam annotation = element.getAnnotation(RequestParam.class);
+
+        if (annotation == null) {
+            return null;
+        }
+
+        String paramName = annotation.name().isEmpty() ? name : annotation.name();
+
+        SlingHttpServletRequest request = InjectorUtils.getSlingHttpServletRequest(adaptable);
+        if (request == null) {
+            return null;
+        }
+
+        if (type instanceof ParameterizedType) {
+            Class<?> collectionType = (Class<?>) ((ParameterizedType) type).getRawType();
+            if (!(ClassUtils.isAssignable(collectionType, Collection.class))) {
+                LOG.debug("RequestParameterInjector doesn't support Collection Type {}", type);
```

Comment:
(cosmetic) Hmm, this phrase is ambiguous) 
If the type is not an instance of Collection, then, it is not a "collection type" by definition). You may say, "what if there's a Set?" But then again, in Java terms, a set is not a collection
So, "doesn't support type..", and that's all